src/modules/job-manager/alloc.c

```diff
@@ -599,14 +599,35 @@ struct job *alloc_queue_next (struct alloc *alloc)
     return zlistx_next (alloc->queue);
 }
 
-/* called from urgency_handle_request() */
+/* called from reprioritize_job() */
 void alloc_queue_reorder (struct alloc *alloc, struct job *job)
 {
     bool fwd = job->priority > (FLUX_JOB_PRIORITY_MAX / 2);
 
     zlistx_reorder (alloc->queue, job->handle, fwd);
 }
 
+int alloc_queue_sort (struct alloc *alloc)
+{
+    struct job *job;
+    zlistx_sort (alloc->queue);
+
+    /*  N.B.: zlistx_sort() invalidates all list handles since
+     *   the sort swaps contents of nodes, not the nodes themselves.
+     *   Therefore, job handles into the list must be re-acquired here:
+     */
+    job = zlistx_first (alloc->queue);
+    while (job) {
+        job->handle = zlistx_cursor (alloc->queue);
+        job = zlistx_next (alloc->queue);
+    }
+
+    if (alloc->mode == SCHED_SINGLE)
```

Comment:
i was a little surprised to find `recalc_pending()` called inside this function. Perhaps rename of function to make less of a surprise? Or perhaps should call separately after `alloc_queue_sort()` is called in `reprioritize_all()`.

Reply:
Can you say why you found it surprising? the comment on `recalc_pending()` says to call it if highest priority job may have changed, which could happen after the list is re-sorted when all priorities may have changed.

Ah, or is your problem with the name `alloc_queue_sort`, which implies it only sorts the queue only?

In that case I agree, the name could be better -- though nothing comes to mind atm

Reply:
Maybe `alloc_queue_reprioritize`? :shrug:

Reply:
> Ah, or is your problem with the name alloc_queue_sort, which implies it only sorts the queue only?

Yeah, mostly the name.  I had assumed it only did the sort.  Given the code above it in `reprioritize_all()`, it made sense you'd want to "finish up" by sorting the queue.    Also `alloc_queue_reorder()` only re-orders the one job.

I think `alloc_queue_reprioritize()` would probably be better with some comment saying it'll sort everything in there.

Or you could go the ugly way, "sort_recalc_pending()`?

Reply:
Ok, I'll change the name and add comment before the next push.